pkg/analyzer/analyzers/airtable/airtablepat/airtable.go

```diff
@@ -0,0 +1,205 @@
+	token, ok := credInfo["token"]
+	if !ok {
+		return nil, errors.New("token not found in credInfo")
+	}
+
+	userInfo, err := common.FetchAirtableUserInfo(token)
+	if err != nil {
+		return nil, err
+	}
+
+	scopeStatusMap[common.PermissionStrings[common.UserEmailRead]] = userInfo.Email != nil
+
+	var basesInfo *common.AirtableBases
+	if granted, err := determineScope(token, common.SchemaBasesRead, nil); granted {
+		if err != nil {
+			return nil, err
+		}
+		basesInfo, err = common.FetchAirtableBases(token)
+		if err != nil {
+			return nil, err
+		}
+		// If bases are fetched, determine the token scopes
+		err := determineScopes(token, basesInfo)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return mapToAnalyzerResult(userInfo, basesInfo), nil
+}
+
+func AnalyzeAndPrintPermissions(cfg *config.Config, token string) {
+	userInfo, err := common.FetchAirtableUserInfo(token)
+	if err != nil {
+		color.Red("[x] Error : %s", err.Error())
+		return
+	}
+
+	scopeStatusMap[common.PermissionStrings[common.UserEmailRead]] = userInfo.Email != nil
+
+	var basesInfo *common.AirtableBases
+	basesReadPermission := common.SchemaBasesRead
+	if granted, err := determineScope(token, basesReadPermission, nil); granted {
+		if err != nil {
+			color.Red("[x] Error : %s", err.Error())
+			return
+		}
+		basesInfo, _ = common.FetchAirtableBases(token)
+		err := determineScopes(token, basesInfo)
+		if err != nil {
+			color.Red("[x] Error : %s", err.Error())
+			return
+		}
+	}
+
+	color.Green("[!] Valid Airtable Personal Access Token\n\n")
+
+	common.PrintUserAndPermissions(userInfo, scopeStatusMap)
+	if scopeStatusMap[common.PermissionStrings[basesReadPermission]] {
+		common.PrintBases(basesInfo)
+	}
+}
+
+func determineScope(token string, scope common.Permission, ids map[string]string) (bool, error) {
+	scopeString := common.PermissionStrings[scope]
+	endpoint := getEndpointByPermission(scopeString)
+	url := endpoint.URL
+	if ids != nil {
+		for _, key := range endpoint.RequiredIDs {
+			if value, ok := ids[key]; ok {
+				url = strings.Replace(url, fmt.Sprintf("{%s}", key), value, -1)
+			}
+		}
+	}
+
+	resp, err := common.CallAirtableAPI(token, endpoint.Method, url)
+	if err != nil {
+		return false, err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode == http.StatusOK {
+		scopeStatusMap[scopeString] = true
+		return true, nil
+	} else if endpoint.ExpectedErrorResponse != nil {
+		var result map[string]interface{}
+		if err := json.NewDecoder(resp.Body).Decode(&result); err != nil {
+			return false, err
+		}
+
+		if errorInfo, ok := result["error"].(map[string]interface{}); ok {
+			if errorType, ok := errorInfo["type"].(string); ok && errorType == endpoint.ExpectedErrorResponse.Type {
+				scopeStatusMap[scopeString] = false
+				return false, nil
+			}
+		}
+	}
+
+	scopeStatusMap[scopeString] = true
+	return true, nil
+}
+
+func determineScopes(token string, basesInfo *common.AirtableBases) error {
+	if basesInfo != nil && len(basesInfo.Bases) > 0 {
```

Comment:
suggestion: Similar to my earlier **"line of sight"** comment, we can invert the logic of this `if` statement to return early, allowing us to de-indent the following `for` loop for better readability.

Ex:
```
if basesInfo == nil || len(basesInfo.Bases) == 0 {
    return nil
}

for _ base := range baseInfo.Bases {
    ...
}
```